client/clip_client/client.py

```diff
@@ -30,26 +30,23 @@ def __init__(self, server: str):
             - port: the public port of the server
         :param server: the server URI
         """
-        try:
-            r = urlparse(server)
-            _port = r.port
-            _scheme = r.scheme
-            if not _scheme:
-                raise
-        except:
+        r = urlparse(server)
+        _port = r.port
+        self.scheme = r.scheme
+        if not self.scheme:
```

Comment:
how to catch the exception during `urlparse(server)`?

Reply:
it would fail either way since we cannot create a connection from a broken link